### PR TITLE
Move sources from cli-ng to common

### LINKED ISSUE #1: Move existing cli-ng code to cs-common
See [common-types](https://github.com/carbynestack/cli-ng/tree/master/src/main/java/io/carbynestack/common) and [common-testing](https://github.com/carbynestack/cli-ng/tree/master/src/test/java/io/carbynestack/testing).

### PR DESCRIPTION
**Pull request by grevend-bosch**
Signed-off-by: David Greven <fixed-term.David.Greven@de.bosch.com>

### CODE REVIEW

**Comment by grevend-bosch** [comment]
@sbckr Blocked until discussed internally.

**Comment by grevend-bosch** [comment]
https://github.com/carbynestack/common/blob/4224fd439a52443ab3457c0d9161e5caeb684c9d/.pre-commit-config.yaml#L8-L12

Needs to be removed because of JDK 17 (see carbynestack/cli-ng#18).

**Comment by grevend-bosch** [comment]
~~@sbckr @strieflin Test coverage can be ignored in this case, as the values ​​cannot be accurately calculated due to the division of classes between common-testing and common-types.~~

File: common-testing/3RD-PARTY-LICENSES/org.apiguardian.apiguardian-api/LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (sbckr) [inline common-testing/3RD-PARTY-LICENSES/org.apiguardian.apiguardian-api/LICENSE]: the location of the file (path) does not comply with the typical convention <groupId>_<artifactId>
`org.apiguardian.apiguardian-api` -> `org.apiguardian_apiguardian-api`

File: common-types/src/main/java/io/carbynestack/common/result/Result.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+     * @return the folded value of mapping either this {@link Success} value
+     * or {@link Failure} reason to the mapping functions bearing a value of
+     * type N
+     * @throws NullPointerException if the failure or success mapping function
+     *                              is {@code null}
+     * @see Failure#reason()
+     * @see Success#value()
+     * @since 0.1.0
+     */
+    <N> N fold(Function<? super F, ? super N> failureFunction, Function<? super S, ? super N> successFunction);
```
Comment (sbckr) [inline common-types/src/main/java/io/carbynestack/common/result/Result.java:156]: Is there a rationale for `fold` taking the `failureFunction` as the first argument and the `successFunction` as the second? This seems rather counterintuitive to me

**Review by sbckr — COMMENTED** [review]
Please see comments on [common-testing/3RD-PARTY-LICENSES/org.apiguardian.apiguardian-api/LICENSE](common-testing/3RD-PARTY-LICENSES/org.apiguardian.apiguardian-api/LICENSE) and [common-types/src/main/java/io/carbynestack/common/result/Result.java](common-types/src/main/java/io/carbynestack/common/result/Result.java)

File: common-types/src/main/java/io/carbynestack/common/result/Result.java (hunk 2 above)
Comment (grevend-bosch) [reply common-types/src/main/java/io/carbynestack/common/result/Result.java:156]: I based the parameter order on similar methods found in other libraries or languages. The main downside is that the defined succession is no longer consistent with the generic parameters.

Comment (grevend-bosch) [reply common-types/src/main/java/io/carbynestack/common/result/Result.java:156]: I definitely understand the confusion and therefore will reverse the order in an upcoming pull request.

**Review by sbckr — APPROVED** [review]
LGTM

Codacy findings on [NOTICE.md](NOTICE.md) ignored as formatting matches our defaults.

**Pull request merged**